Fix errors in staticcontract.h and reader.h

We are getting the following build errors in staticcontract.h and reader.h during win32 debug build:

  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\include\clr\staticcontract.h(326): error C2061: syntax error : identifier 'UINT' (D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\GenIRStubs.cpp) [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\include\clr\staticcontract.h(326): error C2061: syntax error : identifier 'UINT' (D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\reader.cpp) [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\include\clr\staticcontract.h(350): error C2992: 'BlockMarker' : invalid or missing template parameter list (D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\GenIRStubs.cpp) [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\include\clr\staticcontract.h(350): error C2992: 'BlockMarker' : invalid or missing template parameter list (D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\reader.cpp) [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\reader.cpp(6565): error C2079: '__marker_onlyOneAllowedPerScope' uses undefined class 'BlockMarker' [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\reader.cpp(6565): error C2228: left of '.markBlock' must have class/struct/union [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\reader.cpp(6565): error C2228: left of '.useMarkedBlockAnnotation' must have class/struct/union [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\reader.cpp(8088): error C2079: '__marker_onlyOneAllowedPerScope' uses undefined class 'BlockMarker' [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\reader.cpp(8088): error C2228: left of '.markBlock' must have class/struct/union [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\reader.cpp(8088): error C2228: left of '.useMarkedBlockAnnotation' must have class/struct/union [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\include\clr\staticcontract.h(326): error C2061: syntax error : identifier 'UINT' (D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\readerir.cpp) [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\include\clr\staticcontract.h(350): error C2992: 'BlockMarker' : invalid or missing template parameter list (D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\readerir.cpp) [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\readerir.cpp(3188): error C2079: '__marker_onlyOneAllowedPerScope' uses undefined class 'BlockMarker' [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\readerir.cpp(3188): error C2228: left of '.markBlock' must have class/struct/union [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]
  D:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\LLVM\tools\llvm-msilc\lib\Reader\readerir.cpp(3188): error C2228: left of '.useMarkedBlockAnnotation' must have class/struct/union [d:\j\workspace\dotnet_llvm-msilc_windows_debug_win32_prtest\build_debug_win32\tools\llvm-msilc\lib\Reader\msilcReader.vcxproj]

--- a/include/Pal/MSILCPal.h
+++ b/include/Pal/MSILCPal.h
@@ -18,6 +18,9 @@
 // We still need the PAL EH macros on Windows, so define them here.
 #if defined(_MSC_VER)
 
+#if defined(_DEBUG)
+#include <windows.h> // For UINT
+#endif
 #include "staticcontract.h"
 
 // Note: PAL_SEH_RESTORE_GUARD_PAGE is only ever defined in clrex.h, so we only

